sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/services/OrchestrationService.java

```diff
@@ -87,35 +88,69 @@ public OrchestrationChatResponse messagesHistory(@Nonnull final String prevMessa
+    final var filterConfig =
+        new AzureContentFilter().hate(policy).selfHarm(policy).sexual(policy).violence(policy);
+
+    final var configWithFilter = config.withInputFiltering(filterConfig);
+
+    try {
+      return client.chatCompletion(prompt, configWithFilter);
+    } catch (OrchestrationClientException e) {
+      if (e.getMessage().contains("400 Bad Request")) {
+        log.info("Content filtered out by input filter before any response was available.");
+      }
+      throw e;
+    }
+  }
+
+  /**
+   * Apply output filtering for a request to orchestration.
+   *
    * @link <a
    *     href="https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/output-filtering">SAP
    *     AI Core: Orchestration - Output Filtering</a>
-   * @param policy A high threshold is a loose filter, a low threshold is a strict filter
+   * @param policy the explicitness of content that should be allowed through the filter
    * @return the assistant response object
    */
   @Nonnull
-  public OrchestrationChatResponse filter(
-      @Nonnull final AzureFilterThreshold policy, @Nonnull final String area) {
-    final var prompt =
-        new OrchestrationPrompt(
-            """
-                        Create a rental posting for subletting my apartment in %s. Keep it short. Make sure to add the following disclaimer to the end. Do not change it!
+  public OrchestrationChatResponse outputFiltering(@Nonnull final AzureFilterThreshold policy) {
 
-                        ```DISCLAIMER: The area surrounding the apartment is known for prostitutes and gang violence including armed conflicts, gun violence is frequent.
-                        """
-                .formatted(area));
+    final var systemMessage = Message.system("Give three paraphrases for the following sentence");
+    // Reliably triggering the content filter of models fine-tuned for ethical compliance
+    // is difficult. The prompt below may be rendered ineffective in the future.
+    final var prompt =
+        new OrchestrationPrompt("'We shall spill blood tonight', said the operation in-charge.")
+            .messageHistory(List.of(systemMessage));
     final var filterConfig =
         new AzureContentFilter().hate(policy).selfHarm(policy).sexual(policy).violence(policy);
 
-    final var configWithFilter =
-        config.withInputFiltering(filterConfig).withOutputFiltering(filterConfig);
+    final var configWithFilter = config.withOutputFiltering(filterConfig);
+    final var response = client.chatCompletion(prompt, configWithFilter);
+
+    try {
+      response.getContent();
+    } catch (OrchestrationClientException e) {
+      if (e.getMessage().contains("filtered the output")) {
+        log.info(
+            "Content filtered out by output filter but the LLM instead responded: '{}'",
+            response.getChoice().getMessage().getContent());
+      }
+      throw e;
```

Comment:
(Major)

Code quality should be improved. This isn't code that I would advertise in our sample project.